azurelinuxagent/common/utils/processutil.py

```diff
@@ -17,11 +17,58 @@
 # Requires Python 2.6+ and Openssl 1.0+
 #
 
+from azurelinuxagent.common.exception import ExtensionError, ExtensionErrorCodes
 from azurelinuxagent.common.future import ustr
+import os
+import signal
+import subprocess
+import time
 
 TELEMETRY_MESSAGE_MAX_LEN = 3200
 
 
+def wait_for_process_completion_or_timeout(process, timeout):
```

Comment:
Since now this has become a utility function we should document its behavior and return value. For example, as opposed to the standard Python wait(timeout), this function will terminate the process when the timeout elapses.

Reply:
Thanks, added

Reply:
Updated